includes/wf_crm_webform_preprocess.inc

```diff
@@ -179,7 +179,12 @@ class wf_crm_webform_preprocess extends wf_crm_webform_base {
         'scope' => 'footer',
       );
       $page_id = $this->data['contribution'][1]['contribution'][1]['contribution_page_id'];
-      $js_vars['contributionCallback'] = url('civicrm/contribute/transact', array('query' => array('reset' => 1, 'id' => $page_id, 'qfKey' => $this->getQfKey(), 'snippet' => 4), 'alias' => TRUE));
+      $contributionCallbackQuery = array('reset' => 1, 'id' => $page_id, 'qfKey' => $this->getQfKey(), 'snippet' => 4);
+      if ($this->data['contribution'][1]['contribution'][1]['is_test'] == '1') {
+         // RM: This is needed in order for CiviCRM to know that this is a 'test' transaction - otherwise, CiviCRM defaults to 'live' and returns the payment form with public key for the live payment processor!
+        $contributionCallbackQuery['action'] = CRM_Core_Action::description(CRM_Core_Action::PREVIEW); // because CiviCRM treats a transaction with action 'preview' as 'test'
```

Comment:
I don't think Drupal coding standards like comments on the same line as code. Can you move the comment to line above?